src/inmanta/protocol/methods_v2.py

```diff
@@ -463,3 +463,41 @@ def resource_deploy_start(
     :return: A dict mapping the resource version id of each dependency of resource_id to
              the last deployment status of that resource.
     """
+
+
+# No pagination support is provided for this endpoint because there is no elegant way to page the output of this endpoint.
+@typedmethod(
+    path="/resource/<id>/events", operation="GET", arg_options=methods.ENV_OPTS, client_types=[ClientType.api], api_version=2
+)
+def get_resource_events(
+    tid: uuid.UUID,
+    id: model.ResourceVersionIdStr,
+) -> Dict[model.ResourceIdStr, List[model.ResourceAction]]:
+    """
+    Return relevant events for a resource, i.e. all deploy actions for each of its dependencies since this resources' last
+    deploy or all deploy actions if this resources hasn't been deployed before. The resource actions are sorted in descending
+    order according to their started timestamp.
+
+    :param tid: The id of the environment this resource belongs to
+    :param id: The id of the resource to get events for.
+    """
+
+
+@typedmethod(
+    path="/resource/<id>/did_dependency_change",
+    operation="GET",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def resource_should_deploy(
```

Comment:
I think `did_dependency_change` is a better name for the method as well

Nice idea too.

Reply:
Right, I forgot about that in my review. I think the name should at least make clear we're talking about reacting to dependencies' changes. I wasn't sure what would be a good name though.